src/geometry/partitions/distributedptscotchunstructuredpartition.h

```diff
@@ -0,0 +1,291 @@
+
+#ifndef LIBGEODECOMP_GEOMETRY_PARTITIONS_DISTRIBUTEDPTSCOTCHUNSTRUCTUREDPARTITION_H
+#define LIBGEODECOMP_GEOMETRY_PARTITIONS_DISTRIBUTEDPTSCOTCHUNSTRUCTUREDPARTITION_H
+
+#include <libgeodecomp/config.h>
+#include <libgeodecomp/geometry/partitions/partition.h>
+#include <libgeodecomp/geometry/adjacency.h>
+
+#ifdef LIBGEODECOMP_WITH_CPP14
+#ifdef LIBGEODECOMP_WITH_SCOTCH
+#ifdef LIBGEODECOMP_WITH_MPI
+
+#include <mpi.h>
+#include <ptscotch.h>
+#include <chrono>
+
+#ifdef SCOTCH_PTHREAD
+#error can only use ptscotch if compiled without SCOTCH_PTHREAD
+#endif // SCOTCH_PTHREAD
+
+
+namespace LibGeoDecomp {
+
+inline void prettyPrint(std::ostream &os, const std::vector<Region<1>> regions, const Coord<2> dimensions, const Coord<2> &resolution = Coord<2>(16, 16))
+{
+    for (int y = 0; y < resolution.y(); ++y) {
+        for (unsigned int i = 0; i < regions.size(); ++i)
+        {
+            const Region<1> &region = regions[i];
+
+            for (int x = 0; x < resolution.x(); ++x) {
+                int cx = ((float)x / (float)resolution.x()) * dimensions.x();
+                int cy = ((float)y / (float)resolution.y()) * dimensions.y();
+
+                if (region.count(Coord<1>(cy * dimensions.x() + cx)) > 0) {
+                    os << '#';
+                } else {
+                    os << '.';
+                }
+            }
+
+            os << ' ';
+        }
+        os << '\n';
+    }
+}
+
+inline void prettyPrint(std::ostream &os, const Region<2> region, const Coord<2> &resolution = Coord<2>(50, 50))
+{
+    Coord<2> from = region.boundingBox().origin;
+    Coord<2> dim = region.boundingBox().dimensions;
+    for (int y = 0; y < resolution.y(); ++y) {
+        for (int x = 0; x < resolution.x(); ++x) {
+            int cx = from.x() + ((float)x / (float)resolution.x()) * dim.x();
+            int cy = from.x() + ((float)y / (float)resolution.y()) * dim.y();
+
+            if (region.count(Coord<2>(cx, cy)) > 0) {
+                os << '#';
+            } else {
+                os << '.';
+            }
+        }
+        os << '\n';
+    }
+}
+
+template<int DIM>
+class DistributedPTScotchUnstructuredPartition : public Partition<DIM>
+{
+public:
+    using Partition<DIM>::startOffsets;
+    using Partition<DIM>::weights;
+
+    DistributedPTScotchUnstructuredPartition(
+            const Coord<DIM> &origin,
+            const Coord<DIM> &dimensions,
+            const long &offset,
+            const std::vector<std::size_t> &weights,
+            const Adjacency &adjacency) :
+        Partition<DIM>(offset, weights),
+        origin(origin),
+        dimensions(dimensions),
+        numPartitions(weights.size())
+    {
+        this->adjacency = adjacency;
+
+        getStartEnd(MPILayer().rank(), start, end);
+        localCells = end - start;
+        buildRegions();
+    }
+
+    DistributedPTScotchUnstructuredPartition(
+            const Coord<DIM> &origin,
```

Comment:
Please put the space after the ampersand, not before. Our coding style does this only for pointers.